homeassistant/components/blink/alarm_control_panel.py

```diff
@@ -55,8 +55,15 @@ def __init__(self, data, name, sync):
 
     def update(self) -> None:
         """Update the state of the device."""
-        _LOGGER.debug("Updating Blink Alarm Control Panel %s", self._name)
-        self.data.refresh()
+        if self.data.check_if_ok_to_update():
```

Comment:
what does this do?

Reply:
The blink integration was calling refresh multiple times.  This uses a `blinkpy` method to check if an update should be called based on the last refresh time.

https://github.com/fronzbot/blinkpy/blob/a40011cb74e6e110cd8a5330e3879e4ac7a50c79/blinkpy/blinkpy.py#L295